### PR TITLE
Salac/riemann refactor clean

### PR DESCRIPTION
**Pull request by dsalac**
Reduced common code to a single calls that are used across different Riemann solvers.

### CODE REVIEW

File: src/finiteVolume/fluxCalculator/riemannCommon.hpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * @param aR: speed of sound on the right center cell
+ * @param rhoR: density on the right cell center
+ * @param p0R: reference pressure for stiffened gas on right (pass in from EOS). Will be zero if perfect gas or single gas
+ * @param pR: pressure on the right cell center
+ * @param gammaR: specific heat ratio for gas on right (pass in from EOS)
+ * @param pstar0: initial guess at the pressure across contact surface
+ * @param massFlux: mass flux across the cell face(?)
+ * @param p12: interface pressure(?)
+ */
+ablate::finiteVolume::fluxCalculator::Direction riemannSolver(const PetscReal uL, const PetscReal aL, const PetscReal rhoL, const PetscReal p0L, const PetscReal pL, const PetscReal gammaL,
```
Comment (mmcgurn) [inline src/finiteVolume/fluxCalculator/riemannCommon.hpp:23]: For something like we I think we might want to keep riemannSolver function out of the global name space.  Would you mind if we moved it to a static function inside of Riemann?

**Review by mmcgurn — CHANGES_REQUESTED** [review]
Thanks for cleaning this up.  I have just one request for function location.

File: src/finiteVolume/processes/surfaceForce.cpp [hunk 2]
```diff
@@ -4,6 +4,11 @@
 #include "utilities/constants.hpp"
 #include "utilities/mathUtilities.hpp"
 
+#define xexit(S, ...)                                                                                                                                                 \
```
[the remark this answers is not in the document]
Comment (dsalac) [reply src/finiteVolume/processes/surfaceForce.cpp:7]: Dammit. Forgot to remove that.

File: src/finiteVolume/fluxCalculator/riemannCommon.hpp (hunk 1 above)
Comment (dsalac) [reply src/finiteVolume/fluxCalculator/riemannCommon.hpp:23]: This is just to get access to the direction enum. The file riemann.cpp contains the single ideal gas. Is this where you were thinking of moving it to?

Comment (dsalac) [reply src/finiteVolume/fluxCalculator/riemannCommon.hpp:23]: Sure. I can make the Riemann solver a sub-class of fluxCalculator and the three cases sub-classes of the Riemann solver.

**Review by maesemen — APPROVED** [review]

**Review by mmcgurn — APPROVED** [review]
Thanks

**Pull request merged**